components/board/client.go

```diff
@@ -34,6 +36,16 @@ type client struct {
 	info           boardInfo
 	cachedStatus   *commonpb.BoardStatus
 	cachedStatusMu sync.Mutex
+
+	streamCancel  context.CancelFunc
+	streamRunning bool
+	streamReady   bool
+	streamMu      sync.Mutex
+	mu            sync.RWMutex
```

Comment:
I agree and think we should at least move cachedStatus into its own sub-struct with its own mutex, since I want to get rid of the status API anyway.

Reply:
Can i do this change in separate PR?

Reply:
> Can i do this change in separate PR?

Yes